keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/ReshapeVertex.java

```diff
@@ -32,19 +34,12 @@ protected DualNumber dualOp(DualNumber dualNumber) {
         Map<Vertex, PartialDerivatives> reshapedDerivatives = new HashMap<>();
 
         for (Map.Entry<VertexId, DoubleTensor> partialDerivative : derivativeOfOutputsWithRespectToSelf.asMap().entrySet()) {
-            int[] newPartialShape = updatePartialShapeToMatchInput(partialDerivative.getValue().getShape(), inputVertex.getShape());
+            int[] newPartialShape = copyLowRankOverHighRank(partialDerivative.getValue().getShape(), inputVertex.getShape());
```

Comment:
You want to copy to a specific side of the partial shape right? Can the name somehow indicate which side?

Reply:
Yep, I like your `tailEnd` idea